### PR TITLE
refactor: config to tc_cli

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Summary:**  
refactor config to tc_cli

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcall/tree/main/docs

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,25 +45,22 @@ async-graphql-value = "6.0.11"
...
 colored = "2"
 regex = "1.10.2"
 http-cache-reqwest = { version = "0.12.0", features = [
    "manager-moka"
 ], default-features = false }
 reqwest-middleware = "0.2.4"
-async-trait = "0.1.77"
-serde_path_to_error = "0.1.15"
+async-trait = "0.1.76"
+serde_path_to_error = "0.1.14"
```
Comment (amitksingh1490) [inline Cargo.toml:56]:
```suggestion
async-trait = "0.1.77"
serde_path_to_error = "0.1.15"
```

File: tc_core/src/http/client/client_cli.rs [hunk 2]
```diff
@@ -4,14 +4,9 @@ use http_cache_reqwest::{Cache, CacheMode, HttpCache, HttpCacheOptions, MokaMana
...
 
-use super::Response;
-use crate::config::{self, Upstream};
-
-#[async_trait::async_trait]
-pub trait HttpClient: Sync + Send {
-  async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response>;
-  async fn execute_raw(&self, req: reqwest::Request) -> anyhow::Result<reqwest::Response>;
-}
+use super::client::HttpClient;
```
Comment (amitksingh1490) [inline tc_core/src/http/client/client_cli.rs:7]: this needs to go to tc-cli

File: tc_core/src/blueprint/for_config/server.rs [hunk 3]
```diff
@@ -1,146 +1,76 @@
...
-#[serde(rename_all = "camelCase", default)]
-pub struct Batch {
-  pub max_size: usize,
-  pub delay: usize,
-  pub headers: BTreeSet<String>,
-}
-impl Default for Batch {
+impl Default for Server {
   fn default() -> Self {
-    Batch { max_size: 100, delay: 1, headers: BTreeSet::new() }
```
Comment (amitksingh1490) [inline tc_core/src/blueprint/for_config/server.rs:63]: why was this deleted?

File: tc_core/src/blueprint/for_config/server.rs [hunk 4]
```diff
@@ -1,146 +1,76 @@
...
+use hyper::HeaderMap;
 use serde::{Deserialize, Serialize};
 
-use crate::config::{is_default, KeyValues};
+use crate::blueprint::is_default;
 
-#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq)]
-#[serde(rename_all = "camelCase")]
+#[derive(Clone, Debug, Setters)]
 pub struct Server {
```
Comment (amitksingh1490) [inline tc_core/src/blueprint/for_config/server.rs:11]: this is not what was suggested this should not move to bluprint it should be duplicated as did for server

File: src/blueprint/from_config/server.rs [hunk 5]
```diff
@@ -1,142 +0,0 @@
-use std::collections::BTreeMap;
```
Comment (amitksingh1490) [inline src/blueprint/from_config/server.rs:1]: why is this deleted?

File: tc_core/src/http/client/client_cli.rs (hunk 2 above)
Comment (ssddOnTop) [reply tc_core/src/http/client/client_cli.rs:7]: It's used in `request_context` and `server_context`. So I was thinking of separating these based on architecture

like:

```rust
pub mod client;
#[cfg(not(target_arch = "wasm_32"))]
mod client_cli;
#[cfg(target_arch = "wasm_32")]
mod client_wasm;
``` 

So that methods remains same and would be easy to implement, modify and add

File: tc_core/src/http/request_context.rs [hunk 6]
```diff
@@ -39,10 +38,11 @@ pub struct RequestContext {
 
 impl Default for RequestContext {
   fn default() -> Self {
-    let config::Config { server, upstream, .. } = config::Config::default();
+    // let config::Config { server, upstream, .. } = config::Config::default();
     //TODO: default is used only in tests. Drop default and move it to test.
```
[the remark this answers is not in the document]
Comment (ssddOnTop) [reply tc_core/src/http/request_context.rs:42]: .. what needs to be fixed?

[the remark this answers is not in the document]
Comment (amitksingh1490) [reply tc_core/src/http/request_context.rs:42]: //TODO: default is used only in tests. Drop default and move it to test.

[the remark this answers is not in the document]
Comment (ssddOnTop) [reply tc_core/src/http/request_context.rs:42]: I don't think that could be done as RequestContext::default() is now used in `benches/impl_path_string_for_evaluation_context.rs` and `tc_core/src/path.rs`
